### ISSUE TITLE
Add a vcpkg package for the library

### ISSUE THREAD

**Issue by shironecko** [SSBMTonberry/tileson#7]
While Tileson is already pretty easy to integrate, being able to add it as a vcpkg package would be ideal for the projects that are already using vcpkg.

**Comment by SSBMTonberry** [comment]
Hello @shironecko! I have not really used vcpkg before, but I looked up on it, and based on that it is cross platform and is compiling libraries by sources I think this is a great idea! I'll put it on the list for the next version. 

...Also I should remind myself to publish the official 1.1.0 release soon. But the latest commit is probably what will be the 1.1.0 version.

I'll add this feature to the 1.2.0 roadmap!

**Comment by shironecko** [comment]
Hey! Thanks for giving attention to my suggestion!
By the way, are you open to pull requests? I happen to have some free time these days, so I might be able to help with getting the tileson onto vcpkg.

**Comment by SSBMTonberry** [comment]
@shironecko : I am open to pull requests, so feel free to do this task if you have the time :slightly_smiling_face: I have assigned the task to you now! Please mark the commits with #7, so they can get automatically traced to this issue :slightly_smiling_face:

**Comment by SSBMTonberry** [comment]
@shironecko: I presume you didn't get the time to do this task after all, and that is fine :slightly_smiling_face: 
I'm working on developing v1.2.0 of Tileson these days, so I'm re-assigning this task to myself.
If you are partly done or have any details that would be useful to me, please share :slightly_smiling_face:

**Comment by shironecko** [comment]
Sorry, I have created an issue on vcpkg repo a long time ago and kinda forgot about it: https://github.com/microsoft/vcpkg/issues/12035
Thing is, I was only able to find their guidelines on submitting the requests for new packages and no info on what is the actual procedure for doing that is, either I missed something under my nose or the whole thing is more mysterious than it needs to be.

**Comment by SSBMTonberry** [comment]
@shironecko : Thank you for looking into this! So if I understand correctly you have to put a request for the library as a first step, then there is a unknown step of actually linking the library to vcpkg afterwards? I'll look more into this myself when I'm done with the other tasks. Thanks for the URL to the issue :) 

It sure shouldn't have to be a overly complicated process, but it might be. I'm sure we'll figure it out, though, one way or another. I'll post the details in this issue when I find out, eventually!

**Comment by shironecko** [comment]
Yeah, that pretty much sums out the whole thing. It probably is not as complicated as it seems, but there's no documentation that I could find on this topic and the only way is probably just looking through their source code. Which is more difficult than it needs to be because CMake code might as well be Traditional Chinese for me, but that is a rant point and might be a non-issue for CMake proficient people.

**Comment by SSBMTonberry** [comment]
Moving this to the 1.3.0 milestone, as it's considered more a "nice to have", and `1.2.0` has a lot of features that are useful which I'd like to release asap.

**Comment by TheMaverickProgrammer** [comment]
I prefer vcpkg over git submodules for my large projects. Having this lib on vcpkg manager would make it stand out and draw lots of users.

**Comment by SSBMTonberry** [comment]
Thanks for your feedback!
`vcpkg` will happen, but it will be more fitting to have this feature in `1.3.0` nonetheless, as there will be done some work to make the library more mature :slightly_smiling_face: 
Meanwhile: `Tileson` is designed as a header-only and single-header library, so it is easy to include without the use of `submodules`.

**Comment by SSBMTonberry** [comment]
Created an own patch version 1.3.1 to focus on this fully when v1.3.0 is done.
